rancher2/structure_agent_deployment_customization_v2.go

```diff
@@ -21,11 +21,13 @@ func flattenAgentDeploymentCustomizationV2(in *v1.AgentDeploymentCustomization)
 	}
 
 	if in.OverrideAffinity != nil {
-		obj["override_affinity"] = in.OverrideAffinity
+		b, _ := interfaceToJSON(in.OverrideAffinity)
+		_ = b
+		obj["override_affinity"] = b // TODO - ANDY, changed it to keep it simpler instead of creating a new schema.  The expand reads a json
```

Comment:
1. That's fine but please be descriptive with variables. `_` is the error returned from `interfaceToJSON`. We are ignoring this so don't set it. I suggest

```
affinityStr, _ := interfaceToJSON(in.OverrideAffinity)
obj["override_affinity"] = affinityStr
```

2. What were your results of testing a v2 cluster with cluster agent customization defined?

Reply:
1.  That was testing for testing, i wasn't supposed to commit it.   
Current code is: 
```
		b, err := interfaceToJSON(in.OverrideAffinity)
		if err != nil {
			return []interface{}{}, err
		}
		obj["override_affinity"] = b

```

Reply:
Gotcha, go ahead and remove it. Could you also post some brief results of testing a v2 cluster with cluster agent customization? I sent you a test tf config for that :)

Reply:
I'm trying to . . . 
I'm having some issues runing rancher locally from v2.7 and from 2.8 :/  

As soon as I'm able to I'll.

Reply:
2.  I Created an RKE2 cluster with the following cluster_agent_deployment_customization : 

```
  cluster_agent_deployment_customization {
  override_affinity = <<EOF
{
  "nodeAffinity": {
    "requiredDuringSchedulingIgnoredDuringExecution": {
      "nodeSelectorTerms": [{
        "matchExpressions": [{
          "key": "beta.kubernetes.io/os",
          "operator": "NotIn",
          "values": [
            "windows"
          ]
        }]
      }]
    }
  }
}
EOF
    override_resource_requirements {
      cpu_limit      = "800m"
      cpu_request    = "250m"
      memory_limit   = "800Mi"
      memory_request = "250Mi"
    }
  }

```

I wasn't able to make it to be ready (nor I was able to do that using UI). 

But when I checked the management.cattle.io/v3  - Cluster it had: 
 
 
 ```
 spec:
  agentImageOverride: ''
  answers: {}
  clusterAgentDeploymentCustomization:
    overrideAffinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: beta.kubernetes.io/os
                  operator: NotIn
                  values:
                    - windows
    overrideResourceRequirements:
      limits:
        cpu: 800m
        memory: 800Mi
      requests:
        cpu: 250m
        memory: 250Mi

 
 ```

Reply:
@felipe-colussi Could you post those results on the PR description? Thanks!